src/boot/bootctl.c

```diff
@@ -607,6 +613,9 @@ static int boot_config_load_and_select(
         if (r < 0)
                 return r;
 
+        if (arg_root)
```

Comment:
This skips over `boot_config_select_special_entries`, which I think is still relevant with `--root`…

Reply:
refactored